File: SystemDesignDiagrams/Interviews/tic-tac-toe/entities/Board.js
import { Cell } from './Cell.js'

export class Board {
    constructor(size) {
        this.size = size;
        this.board = []
        for (let i = 0; i < size; i++) {
            let rows = []
            for (let j = 0; j < size; j++) {
                rows.push(new Cell(i, j))
            }
            this.board.push(rows)
        }
    }


    updateCell(row, col, player) {
        console.log(player.getSymbol())
        this.board[row][col].updateCell(player.getSymbol())
    }

    getBoard() {
        return this.board;
    }

    printBoard() {
        for(let i = 0; i<this.size; i++) {
            for(let j = 0; j < this.size; j++) {
                console.log(this.board[i][j])
            }
        }
    }

    isFilled() {
        let filled = true;
        for (let i = 0; i < this.size; i++) {
            for (let j = 0; j < this.size; j++) {
                if (!this.board[i][j].getSymbol()) {
                    filled = false
                    break
                }
            }
            if (!filled) {
                break;
            }
        }
        return filled;
    }
}

File: SystemDesignDiagrams/practice/LLD/tictactoe/src/main/java/com/tictactoe/entities/Board.java
package com.tictactoe.entities;

import com.tictactoe.enums.Symbol;

public class Board {
    private final Cell[][] board;
    private final int size;
    public Board(int size) {
        this.size = size;
        this.board = new Cell[size][size];
        for(int i = 0; i < size; i++) {
            for(int j = 0; j < size; j++) {
                board[i][j] = new Cell(i, j);
            }
        }
    }

    public void updateBoard(int row, int col, Symbol symbol) {
        this.board[row][col].updateCell(symbol);
    }

    public int getSize() {
        return this.size;
    }

    public Cell[][] getBoard() {
        return this.board;
    }

    public boolean isFull() {
        boolean isFull = true;
        for(int i = 0; i < size; i++) {
            for(int j = 0; j < size; j++) {
                if(board[i][j].getSymbol() == null) {
                    isFull = false;
                    break;
                }
            }
        }
        return isFull;
    }

    public void printBoard() {
        System.out.println("-------------");
        for (int i = 0; i < size; i++) {
            System.out.print("| ");
            for (int j = 0; j < size; j++) {
                Symbol symbol = board[i][j].getSymbol();
                if(symbol != null) {
                    System.out.print(symbol + " | ");
                }
            }
            System.out.println("\n-------------");
        }
    }
}
